app/services/scheduling_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class SchedulingService
+  DEFAULT_TIME = "9:00am"
+  DEFAULT_DATE = Time.zone.now.tomorrow
+
+  def self.time_options(selected = nil)
+    options = times.map { |time| { text: time, value: time } }
+    if selected
+      options.each do |option|
+        if option[:text] == selected
+          option[:selected] = true
+          break
+        end
+      end
+    else
+      options.each do |option|
+        if option[:text] == DEFAULT_TIME
+          option[:selected] = true
+          break
+        end
+      end
+    end
+    options
+  end
+
+  def self.times
+    hours = [12] + (1..11).to_a
+    incremented_hours = hours.map do |hour|
+      [hour.to_s + ":00", hour.to_s + ":30"]
+    end
+
+    meridiem_hours = incremented_hours.flatten!.map { |hour| hour + "am" }
+    incremented_hours.each { |hour| meridiem_hours << hour + "pm" }
+    meridiem_hours[0] = "12:01am"
```

Comment:
I assume this is from the design, but is there any reason why midnight is special and is at 01 minutes?

Reply:
Yes it's from the design.

Reply:
How weird :-/

Reply:
I think it's something to do with users getting confused by what day 12:00am is in but can check with Sonia when she's back.